src/tests/tribol_enzyme_poly_intersect.cpp

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+//-----------------------------------------------------------------------------
+//
+// file: tribol_enzyme_poly_intersect.cpp
+//
+//-----------------------------------------------------------------------------
+
+#include <iostream>
+
+#include "tribol/config.hpp"
+
+#include "gtest/gtest.h"
+
+#ifdef TRIBOL_USE_UMPIRE
+#include "umpire/ResourceManager.hpp"
+#endif
+
+#include "mfem.hpp"
+
+#include "tribol/common/Enzyme.hpp"
+#include "tribol/geom/GeomUtilities.hpp"
+
+namespace tribol {
+
+/**
+ * @brief Test fixture for the Enzyme-based derivatives of intersection polynomial calculations.
+ */
+class EnzymePolyIntersectTest : public testing::Test {
+ protected:
+  static constexpr double delta_{ 1.0e-8 };
+  void SetUp() override {}
+
+  void CheckIntersection( RealT* x1, RealT* x2, int* stencil_dir, RealT fd_tol = delta_, RealT pos_tol = 1.0e-8,
+                          RealT len_tol = 1.0e-8, [[maybe_unused]] std::string name = "" )
+  {
+    RealT xi[16];
+    OverlapVertexType type[8];
+    IndexT edge1[8];
+    IndexT edge2[8];
+    for ( int i{ 0 }; i < 8; ++i ) {
+      xi[i] = 0.0;
+      xi[i + 8] = 0.0;
+      type[i] = OverlapVertexType::A;
```

Comment:
Looks like OverlapVertexType is new. Perhaps add a comment here for clarity since "A" doesn't tell me anything and would require strong developer familiarity.

Reply:
"A" and "B" are the notation `Intersection2DPolygon()` uses.  I think they should be clarified and made consistent through Tribol, but perhaps in a PR focused on that.  Added a comment for now.